plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/building/N4JSBuildTypeTrackingBuilder.java

```diff
@@ -86,6 +111,69 @@ protected void doBuild(ToBeBuilt toBeBuilt, IProgressMonitor monitor, BuildType
 		}
 	}
 
+	@Inject
+	private ReflectExtensions reflector;
+
+	private QueuedBuildData getQueuedBuildData() {
+		return optimisticGet("queuedBuildData");
+	}
+
+	private IBuilderState getBuilderState() {
+		return optimisticGet("builderState");
+	}
+
+	private <T> T optimisticGet(String fieldName) {
```

Comment:
Why the name "optimisticGet"? :) You are using reflection to retrieve the private fields from the superclass XtextBuilder. reflectGet?